### PR TITLE
feature(auth): add SRP authentication

### PR DESCRIPTION
**Pull request by sungly**
### Description:
- added SRP authentication 
- updated Winston and prompt versioning to fix node incompatibility issues 
- updated readme. 

### How to test locally
`npm run cognito srp-login` and login with your username and password

### CODE REVIEW

File: src/cmds/index.js [hunk 1]
```diff
@@ -28,9 +29,14 @@ const cli = () => {
 
             break;
         case 'login':
-            logger.info('@NOTE: Only user password auth is supported ATM.');
```
Comment (rocellaj) [inline src/cmds/index.js:31]: thoughts of doing another prompt here - or providing option of which type of login ?

File: src/cmds/authentication/srp-login.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import prompt from 'prompt';
+import { SRPClient, calculateSignature, getNowString } from 'amazon-user-pool-srp-client'
+
+import config from '../../config';
+import { hash, logger, cognitoClient } from '../../util';
+
+
+function addSecretHashToParams({ params, username, clientId, clientSecret }) {
```
Comment (rocellaj) [inline src/cmds/authentication/srp-login.js:8]: what do you think of moving this to `util/hash`

File: src/cmds/authentication/srp-login.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            dateNow
+        );
+
+    const challengeResponses = {
+        PASSWORD_CLAIM_SIGNATURE: signatureString,
+        PASSWORD_CLAIM_SECRET_BLOCK: ChallengeParameters.SECRET_BLOCK,
+        TIMESTAMP: dateNow,
+        USERNAME: ChallengeParameters.USERNAME
+      }
+
```
Comment (rocellaj) [inline src/cmds/authentication/srp-login.js:48]: remove extra space.

**Review by rocellaj — COMMENTED** [review]
just a quick review, i can test later with you as well. Just minor comments.

File: src/cmds/index.js (hunk 1 above)
Comment (sungly) [reply src/cmds/index.js:31]: hmmm I'm okay just leaving as is since we're only going to have two types.

Comment (sungly) [reply src/cmds/index.js:31]: we could tackle this on a different ticket?

**Review by rocellaj — APPROVED** [review]
Lgtm

**Pull request merged**